lib/ClangImporter/ImportDecl.cpp

```diff
@@ -322,8 +322,11 @@ getSwiftStdlibType(const clang::TypedefNameDecl *D,
     break;
 
   case MappedCTypeKind::VaList:
-    if (ClangTypeSize != ClangCtx.getTypeSize(ClangCtx.VoidPtrTy))
-      return std::make_pair(Type(), "");
+   if (ClangTypeSize != ClangCtx.getTypeSize(ClangCtx.VoidPtrTy)) {
+      if (ClangCtx.getTargetInfo().getBuiltinVaListKind() !=
+          clang::TargetInfo::AArch64ABIBuiltinVaList)
+        return std::make_pair(Type(), "");
+    }
```

Comment:
I think that the result of the discussion was that we wanted to just simplify this to:

```
    switch (ClangCtx.getTargetInfo().getBuiltinVaListKind()) {
    case clang::TargetInfo::BuiltinVaListKind::CharPtrBuiltinVaList:
    case clang::TargetInfo::BuiltinVaListKind::VoidPtrBuiltinVaList:
    case clang::TargetInfo::BuiltinVaListKind::PowerABIBuiltinVaList:
    case clang::TargetInfo::BuiltinVaListKind::X86_64ABIBuiltinVaList:
    case clang::TargetInfo::BuiltinVaListKind::AAPCSABIBuiltinVaList:
      assert(ClangCtx.getTypeSize(ClangCtx.VoidPtrTy) == ClangTypeSize &&
             "expected va_list type to be sizeof(void *)");
      break;
    case clang::TargetInfo::BuiltinVaListKind::AArch64ABIBuiltinVaList:
      break;
    default: return std::make_pair(Type(), "");
    }
```

Reply:
Yes, we could simplify a lot of things throughout the entire Swift code base, but the purpose of this PR is to add support for the aarch64/Linux platform. Not to edit code affecting the support of any other platforms.

Reply:
This is not a good attitude.  As a contributor, you are responsible for your contribution and its impact on the overall codebase.  We're not asking you to make random unrelated changes, we're asking to make changes that specifically make your contribution — and changes like it — easier to write and maintain.

To put it another way, if you're going to ask every other Swift contributor to make an effort to keep your port functional as they work on the project over the years, we can ask you to put more than the minimum possible effort into your initial patches.

Reply:
I didn't mean to disparage the effort you've put into this port, and I apologize for that.

It has been my experience dealing with this kind of thing — across multiple compilers — that new ports almost inevitably push the codebase in at least one novel way.  Not only is it *acceptable* to take these opportunities to refactor code that affects existing supported platforms, but it's actively *desirable*.  The alternative is that each new port re-learns how to solve the same set of problems and just contributes to an ever-mounting pile of target-specific checks which none of the other maintainers will really understand the purpose of, when in reality there's nearly always some underlying logic that can be applied both across the codebase and across targets.

Minimizing the impact of a patch is appreciated and is usually the right starting point, but it's not a project goal in and of itself to achieve that on a patch-by-patch basis — the project goal is to accomplish something while still ending up with the best code reasonably attainable, and sometimes that's best served by doing more work that ultimately makes your patch simpler and leaves the code in a more maintainable position.  Noticing that kind of opportunity is a large part of the point of getting code review from experienced project maintainers.